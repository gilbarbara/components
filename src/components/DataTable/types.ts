import { CSSProperties, MouseEventHandler, ReactNode } from 'react';
import { Simplify, StringOrNull, StringOrNumber } from '@gilbarbara/types';

import { LoaderType } from '~/components/Loader/types';

import {
  OmitElementProps,
  SortDirection,
  StyledProps,
  VariantWithTones,
  WithAccent,
  WithFlexItem,
  WithLayout,
  WithMargin,
  WithPadding,
  WithRadius,
} from '~/types';

export interface DataTableColumn<T = string> {
  /**
   * Disable sorting for the column.
   * @default false
   */
  disableSort?: boolean;
  /**
   * Hide the column on smaller screens.
   * @default false
   */
  hideOnResponsive?: boolean;
  /**
   * Set the column as an action column and disable sorting.
   * @default false
   */
  isAction?: boolean;
  /**
   * The key to use for the column.
   */
  key: T;
  /**
   * The maximum width for the column.
   */
  max?: StringOrNumber;
  /**
   * The minimum width for the column.
   */
  min?: StringOrNumber;
  /**
   * The fixed width for the column.
   * Overrides `min` and `max`.
   */
  size?: StringOrNumber;
  /**
   * The title for the column.
   */
  title: ReactNode;
}

export type DataTableRowContent = ReactNode | { label: ReactNode; value: string };

export type DataTableRow<T extends string> = Simplify<
  Record<T, DataTableRowContent> & { id?: StringOrNumber }
>;

export interface DataTableKnownProps<T extends string>
  extends StyledProps,
    WithAccent,
    WithFlexItem,
    WithLayout,
    WithMargin,
    WithPadding,
    WithRadius {
  /**
   * Component background color
   * If not set, the component will use 'gray.800' for darkMode otherwise 'gray.50'
   */
  bg?: VariantWithTones;
  /**
   * The breakpoint to switch to responsive mode.
   * @default 768 */
  breakpoint?: number;
  /**
   * Remove background and padding
   * @default false
   */
  clean?: boolean;
  /**
   * The columns to display
   */
  columns: Array<DataTableColumn<T>>;
  /**
   * The data to display
   */
  data: Array<DataTableRow<T>>;
  /**
   * The default column to sort by
   */
  defaultSortColumn?: T;
  /**
   * The default sort direction
   * @default asc
   */
  defaultSortDirection?: SortDirection;
  /**
   * Disable scroll to top on page change
   * @default false
   */
  disableScroll?: boolean;
  /**
   * The size of the loader.
   * @default 128
   */
  loaderSize?: number;
  /**
   * The type of loader to display.
   */
  loaderType?: LoaderType;
  /**
   * Show a loader while data is loading.
   * @default false
   */
  loading?: boolean;
  /**
   * The maximum number of rows to display per page.
   * @default 10 */
  maxRows?: number;
  /**
   * Custom message when no results are found.
   * @default Nothing found
   */
  noResults?: ReactNode;
  onClickPage?: (page: number, totalPages: number) => void;
  onClickSort?: (sortedBy: T, sortDirection: SortDirection) => void;
  /**
   * Show pagination
   * @default true
   */
  pagination?: boolean;
  /**
   * Use remote paginated data.
   */
  remote?: {
    currentPage: number;
    totalPages: number;
    useInternalSorting?: boolean;
  };
  /**
   * Rearrange columns on smaller screens.
   * @default false
   */
  responsive?: boolean;
  /**
   * The duration of the scroll animation on page change.
   * @default 400
   */
  scrollDuration?: number;
  /**
   * The element to scroll to on page change.
   */
  scrollElement?: HTMLElement | null;
  /**
   * The margin to apply when scrolling to the element.
   */
  scrollMargin?: number;
  /**
   * Set the header sticky.
   * @default false
   */
  stickyHeader?: boolean;
  style?: CSSProperties;
  /**
   * The width of the table.
   * @default window.innerWidth */
  width?: number;
}

export type DataTableProps<T extends string = string> = Simplify<
  OmitElementProps<HTMLDivElement, DataTableKnownProps<T>, 'data' | 'wrap'>
>;

export interface DataTableHeadProps extends WithAccent, Pick<DataTableProps, 'clean' | 'columns'> {
  darkMode: boolean;
  isDisabled: boolean;
  isResponsive: boolean;
  onClick: MouseEventHandler;
  sortBy: StringOrNull;
  sortDirection: SortDirection;
  stickyHeader: boolean;
}

export interface DataTableBodyProps
  extends WithAccent,
    Pick<DataTableProps, 'clean' | 'columns' | 'data' | 'loaderSize' | 'loaderType' | 'loading'> {
  darkMode: boolean;
  isResponsive: boolean;
  sortColumn?: string;
}
