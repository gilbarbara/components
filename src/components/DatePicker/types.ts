import { ReactNode } from 'react';
import { DayPickerBase, DayPickerRangeProps, DayPickerSingleProps } from 'react-day-picker';
import { Simplify, StringOrNumber } from '@gilbarbara/types';

import {
  Alignment,
  Color,
  StyledProps,
  WithAccent,
  WithBorder,
  WithBorderless,
  WithDisabled,
  WithMargin,
  WithOpen,
  WithPadding,
  WithRadius,
  WithShadow,
} from '~/types';

export type DatePickerRangeParameter = [from?: string, to?: string];

export type DatePickerRangeClickHandler = (range: DatePickerRangeParameter) => void;
export type DatePickerSingleClickHandler = (isoDate: string) => void;

export interface DatePickerBaseProps extends StyledProps, WithAccent<Color> {
  /**
   * @default Go to today
   */
  currentMonthLabel?: ReactNode;
  fromDate?: string | Date;
  toDate?: string | Date;
}

export interface DatePickerLayoutProps
  extends WithBorder,
    WithMargin,
    WithPadding,
    WithRadius,
    WithShadow {}

export interface DatePickerRangeKnownProps
  extends DatePickerBaseProps,
    DatePickerLayoutProps,
    Omit<DayPickerRangeProps, 'fromDate' | 'mode' | 'selected' | 'onSelect' | 'toDate'> {
  /**
   * The date format to use when displaying the dates.
   * @default en-US
   */
  formatLocale?: string;
  /**
   * For internal use with DatePickerInput
   * @private
   */
  onApply?: (selected: DatePickerRangeParameter) => void;
  onChange?: DatePickerRangeClickHandler;
  readOnly?: boolean;
  /**
   * The initial date.
   */
  selected?: DatePickerRangeParameter;
  /**
   * For internal use with DatePickerInput
   * @private
   */
  showApply?: boolean;
}

export type DatePickerRangeProps = Simplify<DatePickerRangeKnownProps>;

export interface DatePickerSingleKnownProps
  extends DatePickerBaseProps,
    DatePickerLayoutProps,
    Omit<DayPickerSingleProps, 'fromDate' | 'mode' | 'onSelect' | 'selected' | 'toDate'> {
  onChange?: DatePickerSingleClickHandler;
  readOnly?: boolean;
  /**
   * The initial date.
   */
  selected?: string;
}

export type DatePickerSingleProps = Simplify<DatePickerSingleKnownProps>;

export interface DatePickerSelectorBaseProps
  extends DatePickerBaseProps,
    Omit<DayPickerBase, 'disabled' | 'fromDate' | 'mode' | 'selected' | 'toDate'>,
    WithBorderless,
    WithDisabled,
    WithOpen,
    WithMargin {
  /**
   * The date format to use when displaying the dates.
   * @default en-US
   */
  formatLocale?: string;
  large?: boolean;
  /**
   * Whether to show the calendar in range or single mode.
   * @default single
   */
  mode?: 'range' | 'single';
  /**
   * Add a hidden input with the given name.
   */
  name?: string;
  /**
   * For internal use with DatePickerSelector
   * @private
   */
  onApply?: (selected: DatePickerRangeParameter) => void;
  placeholder?: string;
  /**
   * The position of the calendar relative to the input.
   * @default right
   */
  position?: Alignment;
  /**
   * The initial date.
   */
  selected?: DatePickerRangeParameter | string;
  /**
   * The separator for date ranges.
   * @default ' â€” '
   */
  separator?: string;
  /**
   * Show a button to apply the selected range.
   * @default false
   */
  showRangeApply?: boolean;
  /**
   * The width of the selector.
   * It has a min-width of 200px.
   */
  width?: StringOrNumber;
}

export type DatePickerSelectorProps = Simplify<
  DatePickerSelectorBaseProps &
    (
      | {
          mode: 'range';
          onChange?: DatePickerRangeClickHandler;
        }
      | {
          mode: 'single';
          onChange?: DatePickerSingleClickHandler;
        }
    )
>;
