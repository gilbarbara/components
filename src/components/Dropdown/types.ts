import { HiddenInput, Option, Props } from '@gilbarbara/react-dropdown';
import { Simplify, StringOrNumber } from '@gilbarbara/types';

import { StyledProps, WithAccent, WithBorderless, WithMargin, WithOpen } from '~/types';

export type DropdownBaseProps = Omit<
  Props,
  | 'className'
  | 'clearComponent'
  | 'clearOnSelect'
  | 'contentComponent'
  | 'create'
  | 'handleComponent'
  | 'inputComponent'
  | 'loadingComponent'
  | 'menuComponent'
  | 'menuItemComponent'
  | 'noDataComponent'
  | 'onClearAll'
  | 'onCreate'
  | 'onSelectAll'
  | 'optionComponent'
  | 'options'
  | 'searchFn'
  | 'secondaryPlaceholder'
  | 'separatorComponent'
  | 'style'
  | 'styles'
>;

export interface DropdownKnownProps
  extends StyledProps,
    WithAccent,
    WithBorderless,
    WithMargin,
    WithOpen,
    DropdownBaseProps {
  /** @default false */
  allowCreate?: boolean;
  /** @default false */
  closeMultiOnSelect?: boolean;
  /**
   * If set, an input with type hidden will be added to the component with the value of the selected option(s).
   * In case of multiple items, the value will be a string concatenated with "separator".
   */
  inputOptions?: HiddenInput;
  items: Option[];
  large?: boolean;
  /** @default 260 */
  menuMaxHeight?: number;
  onClear?: () => void;
  onCreate?: (item: string, close: () => void) => void;
  onSearch?: (value: string) => void;
  /** @default 260 */
  width?: StringOrNumber;
}

export type DropdownProps = Simplify<DropdownKnownProps>;
