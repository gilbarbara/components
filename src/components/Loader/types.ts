import { Simplify } from '@gilbarbara/types';

import { OmitElementProps, StyledProps, WithBlock, WithColorsDefaultColor } from '~/types';

export type LoaderType = 'grow' | 'pill' | 'pride' | 'pulse' | 'rotate';

export interface LoaderKnownProps
  extends OmitElementProps<
    HTMLDivElement,
    StyledProps & WithBlock & Pick<WithColorsDefaultColor, 'color'>
  > {
  size?: number;
  /** @default pill */
  type?: LoaderType;
}

export type LoaderProps = Simplify<LoaderKnownProps>;
