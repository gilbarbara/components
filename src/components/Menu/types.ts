import { ReactElement } from 'react';
import { Simplify, StringOrNumber } from '@gilbarbara/types';

import {
  OmitElementProps,
  PositionX,
  PositionY,
  StyledProps,
  WithAccent,
  WithChildren,
  WithDisabled,
  WithOpen,
} from '~/types';

export interface MenuKnownProps
  extends StyledProps,
    WithAccent,
    WithChildren,
    WithDisabled,
    WithOpen {
  /**
   * The Menu button.
   * @default An Icon with more-vertical-o
   */
  button?: ReactElement;
  /**
   * Disable closing the menu when you click outside.
   * @default false
   */
  disableCloseOnBlur?: boolean;
  /** @default false */
  disableKeyboardNavigation?: boolean;
  labels?: {
    /** @default Close menu */
    close?: string;
    /**
     * Override the open/close button aria-label.
     */
    name?: string;
    /** @default Open menu */
    open?: string;
  };
  /** @default 200 */
  minWidth?: StringOrNumber;
  onToggle?: (status: boolean) => void;
  /** @default bottom-right */
  position?: PositionX | PositionY;
  /** @default click */
  trigger?: 'click' | 'hover';
}

export type MenuProps = Simplify<OmitElementProps<HTMLDivElement, MenuKnownProps>>;
