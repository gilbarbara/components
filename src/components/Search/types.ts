import { CSSProperties, KeyboardEvent, MouseEvent, ReactNode } from 'react';
import { Simplify, StringOrNumber } from '@gilbarbara/types';

import { Icons, StyledProps, WithAccent, WithBorderless, WithChildren, WithMargin } from '~/types';

export interface SearchItem extends WithAccent {
  label?: ReactNode;
  value: string;
}

export type SearchOnSelect = (event: MouseEvent<HTMLElement> | KeyboardEvent<HTMLElement>) => void;

export interface SearchItemsProps
  extends Required<WithAccent>,
    Pick<SearchProps, 'noResultsLabel'> {
  active: boolean;
  cursor: number;
  height: StringOrNumber;
  isBusy: boolean;
  items: SearchItem[];
  onSelect: SearchOnSelect;
}

export interface SearchItemProps extends Required<WithAccent>, WithChildren {
  isSelected: boolean;
  onSelect: SearchOnSelect;
  value: string;
}

export interface SearchKnownProps extends StyledProps, WithAccent, WithBorderless, WithMargin {
  /**
   * Disable closing the list when you click outside.
   * @default false
   */
  disableCloseOnBlur?: boolean;
  /** @default false */
  disableKeyboardNavigation?: boolean;
  disabled?: boolean;
  height?: StringOrNumber;
  /** @default false */
  hideIcon?: boolean;
  /** @default search */
  icon?: Icons;
  items: SearchItem[];
  loading?: boolean;
  noResultsLabel?: ReactNode;
  onBlur?: (value: string) => void;
  onFocus?: (value: string) => void;
  /**
   * Fires after the "debounce" delay
   */
  onSearch?: (value: string) => void;
  /**
   * Debounce onSearch event in milliseconds.
   * Set it to 0 for real-time events.
   * @default 250
   */
  onSearchDebounce?: number;
  onSelect: (value: string) => void;
  /**
   * Fires immediately when the user types
   */
  onType?: (value: string) => void;
  /** @default Search for... */
  placeholder?: string;
  /**
   * Disable internal filtering.
   * @default false
   */
  remote?: boolean;
  /**
   * Show the list on focus (if available)
   *
   * @default true
   */
  showListOnFocus?: boolean;
  style?: CSSProperties;
  /** @default 100% */
  width?: StringOrNumber;
}

export type SearchProps = Simplify<SearchKnownProps>;
