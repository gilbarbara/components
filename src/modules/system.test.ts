import {
  alignStyles,
  baseStyles,
  borderStyles,
  boxStyles,
  colorStyles,
  dimensionStyles,
  displayStyles,
  flexBoxStyles,
  flexItemStyles,
  getContainerStyles,
  getDisableStyles,
  getOutlineStyles,
  getStyledOptions,
  inputStyles,
  isDarkMode,
  layoutStyles,
  marginStyles,
  paddingStyles,
  positioningStyles,
  radiusStyles,
  shadowStyles,
  textStyles,
} from '~/modules/system';

import { BorderItemSide } from '~/types';

describe('alignStyles', () => {
  it('should return properly', () => {
    expect(alignStyles({ align: 'center' })).toMatchSnapshot();
  });

  it('should return an empty object without the "align" prop', () => {
    expect(alignStyles({})).toEqual({});
  });
});

describe('baseStyles', () => {
  it('should return properly', () => {
    expect(baseStyles({})).toMatchSnapshot();
  });
});

describe('borderStyles', () => {
  it.each<Array<BorderItemSide | true>>([
    [true],
    ['horizontal'],
    ['vertical'],
    ['start'],
    ['end'],
    ['left'],
    ['all'],
  ])(`should return properly for "%p"`, border => {
    expect(borderStyles({ border })).toMatchSnapshot();
  });

  it('should return properly with an array', () => {
    expect(
      borderStyles({
        border: [{ side: 'left', size: 4, color: 'primary.100' }, { side: 'top' }],
      }),
    ).toMatchSnapshot();
  });

  it('should return an empty object without the "border" prop', () => {
    expect(borderStyles({})).toEqual({});
  });
});

describe('boxStyles', () => {
  it('should return properly', () => {
    expect(boxStyles({ bg: 'primary', shadow: 'mid' })).toMatchSnapshot();
  });
});

describe('colorStyles', () => {
  it('should return properly with "background" and "color"', () => {
    expect(colorStyles({ bg: 'primary', color: 'green.100' })).toMatchSnapshot();
  });

  it('should return properly with just "color"', () => {
    expect(colorStyles({ color: 'red.300' })).toMatchSnapshot();
  });

  it('should return an empty object without "color" props', () => {
    expect(colorStyles({})).toEqual({});
  });
});

describe('displayStyles', () => {
  it('should return properly', () => {
    expect(displayStyles({ display: 'flex' })).toMatchSnapshot();
  });

  it('should return an empty object without the "display" prop', () => {
    expect(displayStyles({})).toEqual({});
  });
});

describe('dimensionStyles', () => {
  it('should return properly', () => {
    expect(dimensionStyles({ height: 200, maxWidth: '100%', width: 480 })).toMatchSnapshot();
  });

  it('should return an empty object', () => {
    expect(dimensionStyles({})).toEqual({});
  });
});

describe('flexBoxStyles', () => {
  it('should return properly', () => {
    expect(
      flexBoxStyles({ align: 'center', direction: 'column', gap: 20, wrap: 'wrap' }),
    ).toMatchSnapshot();
  });

  it('should return an empty object without "flexBox" props', () => {
    expect(flexBoxStyles({})).toEqual({});
  });
});

describe('flexItemStyles', () => {
  it('should return properly', () => {
    expect(flexItemStyles({ alignSelf: 'center', flex: true, fill: true })).toMatchSnapshot();
  });

  it('should return properly with flex string', () => {
    expect(flexItemStyles({ alignSelf: 'center', flex: 'grow', fill: true })).toMatchSnapshot();
  });

  it('should return properly with flex object', () => {
    expect(
      flexItemStyles({ alignSelf: 'center', flex: { grow: 1, shrink: 1 }, fill: true }),
    ).toMatchSnapshot();
  });

  it('should return an empty object without "flexItem" props', () => {
    expect(flexItemStyles({})).toEqual({});
  });
});

describe('getContainerStyles', () => {
  it('should return properly', () => {
    const { styles } = getContainerStyles({});

    expect(styles).toMatchSnapshot();
  });

  it('should return properly with "responsive: false"', () => {
    const { styles } = getContainerStyles({}, { responsive: false });

    expect(styles).toMatchSnapshot();
  });

  it('should return properly with "verticalPadding: true"', () => {
    const { styles } = getContainerStyles({}, { verticalPadding: true });

    expect(styles).toMatchSnapshot();
  });
});

describe('getDisableStyles', () => {
  it('should return properly', () => {
    const { styles } = getDisableStyles({});

    expect(styles).toMatchSnapshot();
  });

  it('should return properly with "isButton: true"', () => {
    const { styles } = getDisableStyles({ invert: true }, { isButton: true });

    expect(styles).toMatchSnapshot();
  });

  it('should return properly with "hasPlaceholder: true"', () => {
    const { styles } = getDisableStyles({}, { hasPlaceholder: true });

    expect(styles).toMatchSnapshot();
  });
});

describe('getOutlineStyles', () => {
  it('should return properly', () => {
    expect(getOutlineStyles('#000')).toEqual({
      boxShadow: '0 0 0 3px rgba(0,0,0,0.6)',
      outline: 'none',
      zIndex: 10,
    });
  });
});

describe('getStyledOptions', () => {
  it.each([
    ['onChange', true],
    ['color', false],
  ])('should forward %p: %p', (prop, expected) => {
    const { shouldForwardProp } = getStyledOptions();

    expect(shouldForwardProp(prop)).toBe(expected);
  });
});

describe('inputStyles', () => {
  it('should return custom styles for "input"', () => {
    expect(
      inputStyles({ borderless: true, large: true, prefixSpacing: true }, 'input').styles,
    ).toMatchSnapshot();
  });

  it('should return the default styles for "select"', () => {
    expect(inputStyles({ multiple: false }, 'select').styles).toMatchSnapshot();
  });

  it('should return the default styles for "textarea"', () => {
    expect(inputStyles({ suffixSpacing: true }, 'textarea').styles).toMatchSnapshot();
  });
});

describe('isDarkMode', () => {
  it('should return the default', () => {
    expect(isDarkMode({})).toBe(false);
  });

  it('should return true', () => {
    expect(isDarkMode({ theme: { darkMode: true } })).toBe(true);
  });
});

describe('layoutStyles', () => {
  it('should return properly', () => {
    expect(
      layoutStyles({
        display: 'flex',
        height: 100,
        maxHeight: 80,
        maxWidth: 80,
        minHeight: 20,
        minWidth: 20,
        opacity: 1,
        pointerEvents: 'none',
        transition: 'all 0.2s',
        width: 100,
      }),
    ).toMatchSnapshot();
  });

  it('should return an empty object without "layout" props', () => {
    expect(layoutStyles({})).toEqual({});
  });
});

describe('marginStyles', () => {
  it('should return properly', () => {
    expect(
      marginStyles({
        margin: 'md',
        my: 'xs',
        mx: 'auto',
      }),
    ).toMatchSnapshot();
  });

  it('should return properly for side values', () => {
    expect(
      marginStyles({
        mb: 'lg',
        ml: 'auto',
        mr: 'auto',
        mt: 'xl',
      }),
    ).toMatchSnapshot();
  });

  it('should return an empty object without "margin" props', () => {
    expect(marginStyles({})).toEqual({});
  });
});

describe('paddingStyles', () => {
  it('should return properly', () => {
    expect(
      paddingStyles({
        padding: 'md',
        py: 'xs',
        px: 'lg',
      }),
    ).toMatchSnapshot();
  });

  it('should return properly for side values', () => {
    expect(
      paddingStyles(
        {
          pb: 'lg',
          pl: 'xs',
          pr: 'sm',
          pt: 'xl',
        },
        true,
      ),
    ).toMatchSnapshot();
  });

  it('should return an empty object without "padding" props', () => {
    expect(paddingStyles({})).toEqual({});
  });
});

describe('positioningStyles', () => {
  it('should return properly', () => {
    expect(
      positioningStyles({ bottom: 0, left: 0, right: 0, top: 0, position: 'absolute' }),
    ).toMatchSnapshot();
  });

  it('should return an empty object without "positioning" props', () => {
    expect(positioningStyles({})).toEqual({});
  });
});

describe('radiusStyles', () => {
  it('should return properly with a string', () => {
    expect(radiusStyles({ radius: 'xxs' })).toMatchSnapshot();
  });

  it('should return properly with an object', () => {
    expect(
      radiusStyles({ radius: { bottom: 'xs', left: 'md', right: 'lg', top: 'xl' } }),
    ).toMatchSnapshot();
  });

  it('should return an empty object without the "border-radius" prop', () => {
    expect(radiusStyles({})).toEqual({});
  });
});

describe('shadowStyles', () => {
  const theme = { darkMode: true };

  it.each([
    { name: 'box-shadow (light)', props: { shadow: 'low' as const } },
    { name: 'box-shadow (dark)', props: { theme, shadow: 'mid' as const } },
    { name: 'filter (light)', props: { theme, shadow: 'low' as const }, useFilter: true },
    { name: 'filter (dark)', props: { theme, shadow: 'high' as const }, useFilter: true },
  ])(`should return properly for "$name"`, ({ props, useFilter }) => {
    expect(shadowStyles(props, useFilter)).toMatchSnapshot();
  });

  it('should return an empty object without "shadow" props', () => {
    expect(shadowStyles({})).toEqual({});
  });
});

describe('textStyles', () => {
  it('should return properly', () => {
    expect(
      textStyles({ bold: true, italic: true, size: 'sm', textTransform: 'uppercase' }),
    ).toMatchSnapshot();
  });

  it('should return an empty object without "textStyles" prop', () => {
    expect(textStyles({})).toEqual({});
  });
});
