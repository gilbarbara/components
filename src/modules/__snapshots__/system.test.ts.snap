// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`alignStyles should return properly 1`] = `
{
  "textAlign": "center",
}
`;

exports[`baseStyles should return properly 1`] = `
{
  "boxSizing": "border-box",
  "fontFamily": "inherit",
}
`;

exports[`borderStyles should return properly for ""all"" 1`] = `
{
  "border": "1px solid #e6e6e6",
}
`;

exports[`borderStyles should return properly for ""end"" 1`] = `
{
  "borderInlineEnd": "1px solid #e6e6e6",
}
`;

exports[`borderStyles should return properly for ""horizontal"" 1`] = `
{
  "borderBottom": "1px solid #e6e6e6",
  "borderTop": "1px solid #e6e6e6",
}
`;

exports[`borderStyles should return properly for ""left"" 1`] = `
{
  "borderLeft": "1px solid #e6e6e6",
}
`;

exports[`borderStyles should return properly for ""start"" 1`] = `
{
  "borderInlineStart": "1px solid #e6e6e6",
}
`;

exports[`borderStyles should return properly for ""vertical"" 1`] = `
{
  "borderLeft": "1px solid #e6e6e6",
  "borderRight": "1px solid #e6e6e6",
}
`;

exports[`borderStyles should return properly for "true" 1`] = `
{
  "border": "1px solid #e6e6e6",
}
`;

exports[`borderStyles should return properly with an array 1`] = `
{
  "borderLeft": "4px solid #cdcdfe",
  "borderTop": "1px solid #e6e6e6",
}
`;

exports[`boxStyles should return properly 1`] = `
{
  "map": undefined,
  "name": "1dn1kmt",
  "next": undefined,
  "styles": "
    box-sizing:border-box;font-family:inherit;;
    background-color:#3030e8;color:#ffffff;;
    ;
    ;
    ;
    ;
    ;
    ;
    ;
    ;
    box-shadow:2px 4px 16px 4px rgba(148, 148, 148, 0.24);;
  ",
  "toString": [Function],
}
`;

exports[`colorStyles should return properly with "background" and "color" 1`] = `
{
  "backgroundColor": "#3030e8",
  "border": "1px solid #3030e8",
  "color": "#d3f8d7",
}
`;

exports[`colorStyles should return properly with just "color" 1`] = `
{
  "color": "#ff6666",
}
`;

exports[`displayStyles should return properly 1`] = `
{
  "display": "flex",
}
`;

exports[`flexBoxStyles should return properly 1`] = `
{
  "alignContent": undefined,
  "alignItems": "center",
  "flexDirection": "column",
  "flexWrap": "wrap",
  "gap": "20px",
  "justifyContent": undefined,
  "justifyItems": undefined,
}
`;

exports[`flexItemStyles should return properly 1`] = `
{
  "alignSelf": "center",
  "flex": "1 1",
  "flexBasis": undefined,
  "height": "100%",
  "justifySelf": undefined,
  "order": undefined,
  "width": "100%",
}
`;

exports[`flexItemStyles should return properly with flex object 1`] = `
{
  "alignSelf": "center",
  "flex": "1 1",
  "flexBasis": undefined,
  "height": "100%",
  "justifySelf": undefined,
  "order": undefined,
  "width": "100%",
}
`;

exports[`flexItemStyles should return properly with flex string 1`] = `
{
  "alignSelf": "center",
  "flex": "1 0",
  "flexBasis": undefined,
  "height": "100%",
  "justifySelf": undefined,
  "order": undefined,
  "width": "100%",
}
`;

exports[`getContainerStyles should return properly 1`] = `
"
    padding-left: 16px;
    padding-right: 16px;

    

    @media screen and (min-width: 1024px){padding-left:32px;padding-right:32px;};

    
  "
`;

exports[`getContainerStyles should return properly with "responsive: false" 1`] = `
"
    padding-left: 16px;
    padding-right: 16px;

    

    

    
  "
`;

exports[`getContainerStyles should return properly with "verticalPadding: true" 1`] = `
"
    padding-left: 16px;
    padding-right: 16px;

    
      padding-bottom: 16px;
      padding-top: 16px;
    ;

    @media screen and (min-width: 1024px){padding-left:32px;padding-right:32px;};

    @media screen and (min-width: 1024px){padding-bottom:32px;padding-top:32px;};
  "
`;

exports[`getDisableStyles should return properly 1`] = `
"
    background-color: #f7f7f7;
    border-color: #e6e6e6;
    color: #4d4d4d;
    cursor: not-allowed;

    
  "
`;

exports[`getDisableStyles should return properly with "hasPlaceholder: true" 1`] = `
"
    background-color: #f7f7f7;
    border-color: #e6e6e6;
    color: #4d4d4d;
    cursor: not-allowed;

    
      &::placeholder {
        color: #b3b3b3;
      }
    ;
  "
`;

exports[`getDisableStyles should return properly with "isButton: true" 1`] = `
"
    background-color: #f7f7f7;
    border-color: #e6e6e6;
    color: #808080;
    cursor: not-allowed;

    
  "
`;

exports[`inputStyles should return custom styles for "input" 1`] = `
"
    color: #101010;
    display: block;
    font-family: inherit;
    font-size: 16px;
    height: 48px;
    line-height: 1.4;
    padding: 16px 0 16px 40px;
    width: 100%;
    
        background-color: transparent;
        border: 0;
        border-bottom: 1px solid #808080;
      ;

    &:focus {
      box-shadow: 0 3px 0 0 rgba(48,48,232,0.5);
      
      outline: none;
    }

    &:disabled {
      
    background-color: ;
    border-color: #e6e6e6;
    color: #4d4d4d;
    cursor: not-allowed;

    
      &::placeholder {
        color: #b3b3b3;
      }
    ;
  ;
    }

    
      &::placeholder {
        color: #808080;
      }

      &:not(:placeholder-shown) {
        border-color: #3030e8;
      }

      &:read-only {
        
    background-color: ;
    border-color: #e6e6e6;
    color: #4d4d4d;
    cursor: not-allowed;

    
      &::placeholder {
        color: #b3b3b3;
      }
    ;
  ;
      }
    ;
  "
`;

exports[`inputStyles should return the default styles for "select" 1`] = `
"
    color: #101010;
    display: block;
    font-family: inherit;
    font-size: 16px;
    height: 40px;
    line-height: 1.4;
    padding: 8px 24px 8px 16px;
    width: 100%;
    
        background-color: #fff;
        border: 1px solid #808080;
        border-radius: 8px;
      ;

    &:focus {
      
      box-shadow:0 0 0 3px rgba(48,48,232,0.6);outline:none;z-index:10;
      outline: none;
    }

    &:disabled {
      
    background-color: #f7f7f7;
    border-color: #e6e6e6;
    color: #4d4d4d;
    cursor: not-allowed;

    
      &::placeholder {
        color: #b3b3b3;
      }
    ;
  ;
    }

    
  "
`;

exports[`inputStyles should return the default styles for "textarea" 1`] = `
"
    color: #101010;
    display: block;
    font-family: inherit;
    font-size: 16px;
    height: ;
    line-height: 1.4;
    padding: 8px 40px 8px 16px;
    width: 100%;
    
        background-color: #fff;
        border: 1px solid #808080;
        border-radius: 8px;
      ;

    &:focus {
      
      box-shadow:0 0 0 3px rgba(48,48,232,0.6);outline:none;z-index:10;
      outline: none;
    }

    &:disabled {
      
    background-color: #f7f7f7;
    border-color: #e6e6e6;
    color: #4d4d4d;
    cursor: not-allowed;

    
      &::placeholder {
        color: #b3b3b3;
      }
    ;
  ;
    }

    
      &::placeholder {
        color: #808080;
      }

      &:not(:placeholder-shown) {
        border-color: #3030e8;
      }

      &:read-only {
        
    background-color: #f7f7f7;
    border-color: #e6e6e6;
    color: #4d4d4d;
    cursor: not-allowed;

    
      &::placeholder {
        color: #b3b3b3;
      }
    ;
  ;
      }
    ;
  "
`;

exports[`layoutStyles should return properly 1`] = `
{
  "display": "flex",
  "height": "100px",
  "maxHeight": "80px",
  "maxWidth": "80px",
  "minHeight": "20px",
  "minWidth": "20px",
  "opacity": 1,
  "pointerEvents": "none",
  "transition": "all 0.2s",
  "width": "100px",
}
`;

exports[`marginStyles should return properly 1`] = `
{
  "margin": "16px",
  "marginBottom": "8px",
  "marginLeft": "auto",
  "marginRight": "auto",
  "marginTop": "8px",
}
`;

exports[`marginStyles should return properly for side values 1`] = `
{
  "marginBottom": "24px",
  "marginLeft": "auto",
  "marginRight": "auto",
  "marginTop": "32px",
}
`;

exports[`paddingStyles should return properly 1`] = `
{
  "padding": "16px",
  "paddingBottom": "8px",
  "paddingLeft": "24px",
  "paddingRight": "24px",
  "paddingTop": "8px",
}
`;

exports[`paddingStyles should return properly for side values 1`] = `
{
  "paddingBottom": "24px !important",
  "paddingLeft": "8px !important",
  "paddingRight": "12px !important",
  "paddingTop": "32px !important",
}
`;

exports[`positioningStyles should return properly 1`] = `
{
  "bottom": "0px",
  "left": "0px",
  "position": "absolute",
  "right": "0px",
  "top": "0px",
}
`;

exports[`radiusStyles should return properly with a string 1`] = `
{
  "borderRadius": "4px",
}
`;

exports[`radiusStyles should return properly with an object 1`] = `
{
  "borderBottomLeftRadius": "16px",
  "borderBottomRightRadius": "8px",
  "borderTopLeftRadius": "16px",
  "borderTopRightRadius": "24px",
}
`;

exports[`shadowStyles should return properly for "box-shadow (dark)" 1`] = `
{
  "boxShadow": "2px 4px 16px 4px rgba(222, 222, 222, 0.24)",
}
`;

exports[`shadowStyles should return properly for "box-shadow (light)" 1`] = `
{
  "boxShadow": "2px 4px 16px 2px rgba(148, 148, 148, 0.16)",
}
`;

exports[`shadowStyles should return properly for "filter (dark)" 1`] = `
{
  "filter": "drop-shadow(2px 4px 18px rgba(222, 222, 222, 0.8))",
}
`;

exports[`shadowStyles should return properly for "filter (light)" 1`] = `
{
  "filter": "drop-shadow(2px 4px 8px rgba(222, 222, 222, 0.4))",
}
`;

exports[`textStyles should return properly 1`] = `
{
  "fontSize": "14px",
  "fontStyle": "italic",
  "fontWeight": 700,
  "lineHeight": 1.2,
  "textTransform": "uppercase",
}
`;
